### DISCUSSION TITLE
How to improve Timelapse quality

### DISCUSSION THREAD

**Discussion by dbainbridge** [aaronwmorris/indi-allsky#661] [Q&A]
What configuration do I need to change in order to drastically improve the quality of the mp4 Timelapse nightly video?  It currently is too grainy especially when it catches the aurora.

**Comment by aaronwmorris** [comment]
Remind me what camera and settings you currently use?

**Reply by dbainbridge** [reply]
ZWO 178MC.  The jpeg images look really good but video looks crappy.

**Comment by dbainbridge** [comment]
I already have the FFMPEG bitrate set to 5000k and see artifacts like this:

<img width="734" alt="Screenshot 2023-03-24 at 9 16 59 AM" src="https://user-images.githubusercontent.com/881864/227547987-a79d5389-1c43-4870-b373-95fa637fed39.png">

**Reply by aaronwmorris** [reply]
Does it clean up later in the video?

**Reply by dbainbridge** [reply]
No, because it is mostly Aurora so looks "awful" (at least compared to original images)..  https://vimeo.com/811295758.  When there isn't aurora it normally looks good and only have that initial blockiness like you are talking about at the beginning and then cleans up later.  But with aurora all night it never recovers.

**Reply by aaronwmorris** [reply]
Yeah, the same thing happens when you have clouds with almost no contrast or details.  The FFMPEG settings I have implemented are terrible in that condition.  Let me do some investigation.

**Reply by dbainbridge** [reply]
I'm trying -crf 20 and got rid of the other bitrate specification line and regenerating now.

**Comment by dbainbridge** [comment]
I changed it to this: 
`        cmd = [
            'ffmpeg',
            '-y',
            '-loglevel', 'level+warning',
            '-f', 'image2',
            '-r', '{0:d}'.format(self.config['FFMPEG_FRAMERATE']),
            #'-start_number', '0',
            #'-pattern_type', 'glob',
            '-i', '{0:s}/%05d.{1:s}'.format(str(self.seqfolder_p), self.config['IMAGE_FILE_TYPE']),
            '-vcodec', '{0:s}'.format(self.config['FFMPEG_CODEC']),
            '-crf', '20',
            '-pix_fmt', 'yuv420p',
            '-movflags', '+faststart',
        ]`

But did not notice any difference.  Verified the timestamp of the mp4 file was updated.  File size is near same.

**Reply by dbainbridge** [reply]
Can't seem to get ANY improvement.. for some reason with options I've tried.

**Reply by dbainbridge** [reply]
What do I need to restart when I modify Timelapse.py?  Isn't wasn't taking my changes since I needed to restart something.  Rebooting does but not sure what minimum I need to do for it to pick up my changes.

**Reply by aaronwmorris** [reply]
You have to restart the entire indi-allsky service for the new python code to go into effect.

```
systemctl --user restart indi-allsky
```

When you are doing testing, its usually easiest to just run the service from the command line.

```
source virtualenv/indi-allsky/bin/activate

./allsky.py run
```

**Comment by aaronwmorris** [comment]
I tried `-qscale:v 15` with really good results.  5, 15, 25, 31 all resulted in the same file size.  It took 40 minutes to generate and was 10x larger than normal... but it looked good.

I have been reading that the order of the parameters for ffmpeg is important.  I will play with it some more.

**Comment by aaronwmorris** [comment]
Looks like you just have to really bump up the bit rate.  `20000k` shows a significant improvement similar to the qscale option.  I am testing with a 290mm and 385mc, which are only 1k cameras.  The files are much bigger, but show almost no noticeable compression artifacts.

In your case with the 178mc, which is basically a 4k camera, you may even consider `40000k` (or higher) if you are outputting native resolution.

**Reply by aaronwmorris** [reply]
I would really like to see that video with the enhanced bitrate.

**Comment by aaronwmorris** [comment]
Did you ever get a chance to regenerate that timelapse with the increased bitrate?
